OpenRA.Mods.AS/Traits/TeleportNetwork.cs

```diff
@@ -0,0 +1,95 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2015- OpenRA.Mods.AS Developers (see AUTHORS)
+ * This file is a part of a third-party plugin for OpenRA, which is
+ * free software. It is made available to you under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation. For more information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.AS.Traits
+{
+	// TO-DO: Create a proper check for Types of TeleportNetwork and TeleportNetworkManager or lint rule.
+	[Desc("This actor can teleport actors like Nydus canels in SC1. Assuming static object. The actor must have PrimaryBuilding trait so that exit can be specified!")]
+	public class TeleportNetworkInfo : ITraitInfo
+	{
+		[Desc("Type of TeleportNetwork that pairs up, in order for it to work.")]
+		public HashSet<string> Types = new HashSet<string>() { "default" };
```

Comment:
Do we need these to be a hashset, come to think of it? An 1-1 relation (with both being a singular string) might be enough. Feel free coming up with an usecase for the N-N relation though.